compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/JvmCreateTypeGen.java

```diff
@@ -0,0 +1,1720 @@
+                    mv.visitTypeInsn(NEW, className);
+                    mv.visitInsn(DUP_X1);
+                    mv.visitInsn(SWAP);
+                    mv.visitInsn(ACONST_NULL);
+                    String descriptor = String.format("(L%s;[L%s;)V", TYPE, MAP_VALUE);
+                    mv.visitMethodInsn(INVOKESPECIAL, className, JVM_INIT_METHOD, descriptor, false);
+                    String fieldType = String.format("L%s;", TYPEDESC_VALUE);
+                    mv.visitFieldInsn(PUTSTATIC, typeOwnerClass, jvmTypeGen.getTypedescFieldName(name), fieldType);
+
+                    break;
+                case TypeTags.OBJECT:
+                    createObjectType(mv, (BObjectType) bType);
+                    break;
+                case TypeTags.ERROR:
+                    createErrorType(mv, (BErrorType) bType, bType.tsymbol.name.value);
+                    break;
+                case TypeTags.TUPLE:
+                    createTupleType(mv, (BTupleType) bType);
+                    break;
+                default:
+                    createUnionType(mv, (BUnionType) bType);
+                    break;
+            }
+
+            mv.visitFieldInsn(PUTSTATIC, typeOwnerClass, getTypeFieldName(name), String.format("L%s;", TYPE));
+            bTypesCount++;
+            if (bTypesCount % MAX_TYPES_PER_METHOD == 0) {
+                mv.visitInsn(RETURN);
+                mv.visitMaxs(0, 0);
+                mv.visitEnd();
+            }
+        }
+        if (methodCount != 0 && bTypesCount % MAX_TYPES_PER_METHOD != 0) {
+            mv.visitInsn(RETURN);
+            mv.visitMaxs(0, 0);
+            mv.visitEnd();
+        }
+
+        return methodCount;
+    }
+    private List<String> populateTypes(ClassWriter cw, List<BIRTypeDefinition> typeDefs, String typeOwnerClass,
+                                              SymbolTable symbolTable) {
+
+        List<String> funcNames = new ArrayList<>();
+        String fieldName;
+        for (BIRTypeDefinition optionalTypeDef : typeDefs) {
+            BType bType = optionalTypeDef.type;
+            if (!(bType.tag == TypeTags.RECORD || bType.tag == TypeTags.ERROR || bType.tag == TypeTags.OBJECT
+                    || bType.tag == TypeTags.UNION || bType.tag == TypeTags.TUPLE)) {
+                continue;
+            }
+
+            fieldName = getTypeFieldName(optionalTypeDef.internalName.value);
+            String methodName = String.format("$populate%s", fieldName);
+            funcNames.add(methodName);
+
+            MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, methodName, "()V", null, null);
+            mv.visitCode();
+            mv.visitFieldInsn(GETSTATIC, typeOwnerClass, fieldName, String.format("L%s;", TYPE));
+
+            BIRVarToJVMIndexMap indexMap = new BIRVarToJVMIndexMap();
+            switch (bType.tag) {
+                case TypeTags.RECORD:
+                    BRecordType recordType = (BRecordType) bType;
```

Comment:
Shall we have separate methods for each types and call them in this switch case?